de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/api/impl/PrintJobBL.java

```diff
@@ -166,10 +199,73 @@ public int createPrintJobs(final IPrintingQueueSource source,
 		finally
 		{
 			batchMonitor.finish();
+			enqueueForPDFPrinting(source, monitor, pjis, printJobCount);
+			
 		}
 
 		return printJobCount;
 	}
+	
+	private void enqueueForPDFPrinting(final IPrintingQueueSource source, final IPrintJobMonitor monitor, final List<I_C_Print_Job_Instructions> pjis , final int printJobCount)
+	{
+		if (pjis.size() > 0)
+		{
```

Comment:
here too: instead of nesting the while method into an if, why not do something like 
```java
if(pjis.isEmpty())
{
   return;
}